src/main/java/bio/terra/app/controller/CohortBuilderApiController.java

```diff
@@ -0,0 +1,432 @@
+                                        Arrays.asList(
+                                            new DisplayHintEnumEnumHintValues()
+                                                .enumVal(
+                                                    new ValueDisplay()
+                                                        .display("Hispanic or Latino")
+                                                        .value(
+                                                            new Literal()
+                                                                .dataType(DataType.INT64)
+                                                                .valueUnion(
+                                                                    new LiteralValueUnion()
+                                                                        .int64Val(38003563L))))
+                                                .count(54453),
+                                            new DisplayHintEnumEnumHintValues()
+                                                .enumVal(
+                                                    new ValueDisplay()
+                                                        .display("Not Hispanic or Latino")
+                                                        .value(
+                                                            new Literal()
+                                                                .dataType(DataType.INT64)
+                                                                .valueUnion(
+                                                                    new LiteralValueUnion()
+                                                                        .int64Val(38003564L))))
+                                                .count(2272403))))))
+            .addDisplayHintsItem(
+                new DisplayHint()
+                    .attribute(
+                        new Attribute()
+                            .name("year_of_birth")
+                            .type(Attribute.TypeEnum.SIMPLE)
+                            .dataType(DataType.INT64))
+                    .displayHint(
+                        new DisplayHintDisplayHint()
+                            .numericRangeHint(
+                                new DisplayHintNumericRange().min(1909.0D).max(1983.0D))));
+
+    return new ResponseEntity<>(new DisplayHintList(), HttpStatus.NOT_IMPLEMENTED);
+  }
+
+  // Our stubs have the same count for everything
+  private List<InstanceCount> createInstanceCountsYearGenderRace(
+      List<ValueDisplay> genders, List<ValueDisplay> races, List<ValueDisplay> yearsOfBirth) {
+    List<InstanceCount> instanceCounts = new ArrayList<>();
+    genders.forEach(
+        (ValueDisplay gender) ->
+            races.forEach(
+                race ->
+                    yearsOfBirth.forEach(
+                        (ValueDisplay yearOfBirth) -> {
+                          instanceCounts.add(
```

Comment:
Again for future reference, you can use streaming to do this too. If you have nested streaming operations like this, the key is to use `flatMap` to create a flattened stream of the innermost type, which you can then call `map` on.  Although in this case I think you'd need to generate some intermediate tuple types which is a bit awkward:

```suggestion
    record GR(ValueDisplay gender, ValueDisplay race) { } 
    record GRY(ValueDisplay gender, ValueDisplay race, ValueDisplay year) { }
    List<InstanceCount> instanceCounts =
    genders.stream().flatMap(gender -> races.stream().map(race -> new GR(gender, race)))
      .flatMap(gr -> yearsOfBirth.stream().map(yearOfBirth -> new GRY(gr.gender, gr.race, yearOfBirth)))
      .map(gry -> ...)
```

Reply:
The records are neat, thank you! I was having trouble getting the flatmaps to work.